### PR TITLE
Add api paddle.device.cuda.empty_cache to release idle gpu memory hold by allocator。

### PR DESCRIPTION
**Pull request by xiaolao**
### PR types
New features

### PR changes
APIs

### Describe
Add API paddle.device.cuda.empty_cache to releases idle cached memory held by the allocator so that those can be used in other GPU application and visible in `nvidia-smi`. In most cases you don't need to use this function, Paddle does not release the memory back to the OS when you remove Tensors on the GPU, Because it keeps gpu memory in a pool so that next allocations can be done much faster.

exaples:

```python
            import paddle

            # required: gpu
            paddle.set_device("gpu")
            # nvidia-smi
            tensor = paddle.randn([512, 512, 512], "float")
            del tensor
            # nvidia-smi
            paddle.device.cuda.empty_cache()
```

### CODE REVIEW

File: python/paddle/device/cuda/__init__.py [hunk 1]
```diff
@@ -117,3 +118,27 @@ def device_count():
...
+
+def empty_cache():
+    """
+    Releases idle cached memory held by the allocator so that those can be used in other GPU
+    application and visible in `nvidia-smi`. In most cases you don't need to use this function,
+    Paddle does not release the memory back to the OS when you remove Tensors on the GPU,
+    Because it keeps gpu memory in a pool so that next allocations can be done much faster.
+
+    Examples:
+        .. code-block:: python
```
Comment (dingjiaweiww) [inline python/paddle/device/cuda/__init__.py:131]: .. code-block:: python 和import paddle 中间要隔一行

**Review by XiaoguangHu01 — DISMISSED** [review]
LGTM

**Review by sneaxiy — DISMISSED** [review]
LGTM.

**Review by TCChenlong — DISMISSED** [review]

**Review by jzhang533 — DISMISSED** [review]
lgtm

File: python/paddle/device/cuda/__init__.py [hunk 2]
```diff
@@ -117,3 +118,28 @@ def device_count():
...
+
+            import paddle
+
+            # required: gpu
+            paddle.set_device("gpu")
+            # nvidia-smi
```
Comment (jzhang533) [inline python/paddle/device/cuda/__init__.py]: 示例代码里写`nvidia-smi`含义并不清楚呀。

Comment (xiaolao) [reply python/paddle/device/cuda/__init__.py]: 已将 `nvidia-smi` 注释删除

**Review by jzhang533 — APPROVED** [review]
LGTM

**Review by lanxianghit — APPROVED** [review]
LGTM for API change

**Pull request merged**